source/plugins/go/input/lib/cadvisor.go

```diff
@@ -230,14 +230,23 @@ func getResponse(winNode map[string]string, relativeUri string) (*http.Response,
 }
 
 func GetMetrics(winNode map[string]string, namespaceFilteringMode string, namespaces []string, metricTime string) []metricDataItem {
+	defer func() {
+		if r := recover(); r != nil {
+			stacktrace := debug.Stack()
+			Log.Printf("GetMetrics: PANIC RECOVERED: %v, stacktrace: %s", r, stacktrace)
+			SendException(fmt.Sprintf("Error: %v, stackTrace: %v", r, stacktrace))
```

Comment:
I don't exactly know what exception can be generated from here but do we need to send some contextual information to understand what caused this exception?

Reply:
the stacktrace will contain the contextual info but you are right let me add the function name here as well